apis/datadoghq/v1alpha1/datadogmonitor_types.go

```diff
@@ -20,6 +20,10 @@ type DatadogMonitorSpec struct {
 	Priority int64 `json:"priority,omitempty"`
 	// Query is the Datadog monitor query
 	Query string `json:"query,omitempty"`
+	// RestrictedRoles is a list of unique role identifiers to define which roles are allowed to edit the monitor.
+	// `restricted_roles` is the successor of `locked`. For more information about `locked` and `restricted_roles`,
+	// see the [monitor options docs](https://docs.datadoghq.com/monitors/guide/monitor_api_options/#permissions-options).
+	RestrictedRoles []string `json:"restricted_roles,omitempty"`
```

Comment:
nit - too late not but splitting this seemingly unrelated changes into two PRs would make reviews faster